### PR TITLE
Fix Collaborations and edit threads

### PR DESCRIPTION
**Pull request by sha14r**
## Description
Fixes: https://www.notion.so/hicommonwealth/Collaborations-doesn-t-show-new-collaborators-or-let-them-fix-edit-thread-a7728238d48c45919d42703cc258d17b

## How has this been tested?

I used 2 users. One admin and one member. 

Admin (address ID: 22478) creates a thread:  
Thread ID: 3727

Member(address ID: 22482) creates a thread:
Thread ID: 3728

image

Admin is able to edit both threads:
image
image

Member is only able to edit the thread they authored:

image

image

Entry in the Collaborations table:
image

## Have proper tags been added (for bug, enhancement, breaking change)?
- [ ] yes

## Does this PR affect any server routes?
- [ ] yes, and they are tested: [enter the % coverage here]
- [ ] yes, and they are not tested: [enter the % coverage here]
- [ ] yes, but I did not run tests
- [ ] no

### CODE REVIEW

File: server/routes/editThread.ts [hunk 1]
```diff
@@ -35,47 +49,57 @@ const editThread = async (models: DB, req: Request, res: Response, next: NextFun
...
 
   let thread;
   const userOwnedAddresses = await req.user.getAddresses();
-  const userOwnedAddressIds = userOwnedAddresses.filter((addr) => !!addr.verified).map((addr) => addr.id);
-  const collaboration = await models.Collaboration.findOne({
+  const userOwnedAddressIds = userOwnedAddresses
+    .filter((addr) => !!addr.verified)
+    .map((addr) => addr.id);
+
+  const moderator = await models.Role.findOne({
```
Comment (sha14r) [inline server/routes/editThread.ts:82]: Changes here

File: server/routes/editThread.ts [hunk 2]
```diff
@@ -35,47 +49,57 @@ const editThread = async (models: DB, req: Request, res: Response, next: NextFun
...
     where: {
       chain_id: chain.id,
       address_id: { [Op.in]: userOwnedAddressIds },
-      permission: 'admin'
-    }
+      permission: 'admin',
+    },
   });
 
-  if (collaboration || admin) {
```
Comment (sha14r) [inline server/routes/editThread.ts:74]: Changes here

File: server/routes/editThread.ts [hunk 3]
```diff
@@ -129,14 +153,32 @@ const editThread = async (models: DB, req: Request, res: Response, next: NextFun
     await thread.save();
     await attachFiles();
 
+    const collaboration = await models.Collaboration.findOne({
```
Comment (sha14r) [inline server/routes/editThread.ts:156]: Changes here

File: server/routes/editThread.ts [hunk 4]
```diff
@@ -35,47 +49,57 @@ const editThread = async (models: DB, req: Request, res: Response, next: NextFun
...
-  if (collaboration || admin) {
+  if (admin || moderator) {
     thread = await models.OffchainThread.findOne({
       where: {
-        id: thread_id
-      }
+        id: thread_id,
+      },
     });
   } else {
```
Comment (sha14r) [inline server/routes/editThread.ts:104]: The condition to check if the user is the author already exists here

**Comment by jnaviask** [comment]
As far as I can tell, this PR _removes all collaborators_ functionality, rather than resolving the problem, which was that non-moderator/non-admin users marked as collaborator could not edit the threads they were marked on.

**Pull request closed without merging**